v1/test/Registry.js

```diff
@@ -0,0 +1,139 @@
+const { didContractThrow } = require("../../common/SolidityTestUtils.js");
+
+const truffleAssert = require("truffle-assertions");
+
+const Registry = artifacts.require("Registry");
+
+contract("Registry", function(accounts) {
+  // A deployed instance of the Registry contract, ready for testing.
+  let registry;
+
+  const governance = accounts[0];
+  const writer = accounts[1];
+  const creator1 = accounts[2];
+  const creator2 = accounts[3];
+  const rando1 = accounts[4];
+  const rando2 = accounts[5];
+
+  // Corresponds to Registry.Roles.
+  const governanceRole = "0";
+  const writerRole = "1";
+  const derivativeCreatorRole = "2";
+
+  beforeEach(async function() {
+    registry = await Registry.new();
+    await registry.resetMember(writerRole, writer);
+  });
+
+  const areAddressesEqual = (address1, address2) => {
+    return address1.toLowerCase() === address2.toLowerCase();
+  };
+
+  it("Derivative creation", async function() {
+    // No creators should be registered initially.
+    assert.isNotTrue(await registry.holdsRole(derivativeCreatorRole, creator1));
```

Comment:
Can you think of a way to abstract the permissioning model part out of the old tests and make the test bodies just a function that we put in `common/`? May be more difficult than it's worth, but just wanted to propose the idea to see if you think it would make maintaining tests easier.

Reply:
Not super easily for this test case. However, we might be able to move the last two test cases out though, since they exercise other parts of the functionality. I don't know if it's worth it though, I'd almost expect this version of the Registry to evolve independently. E.g., I could see a method to add a new party to a derivative.

Reply:
Yeah, agreed. This test is probably too small to warrant the effort. In the meantime, we'll think more about how to easily share test functionality or do versioning better so we have a plan when we have to start touching the tokenized derivative.